python/mlx/nn/layers/dropout.py

```diff
@@ -32,4 +32,54 @@ def __call__(self, x):
 
         mask = mx.random.bernoulli(self._p_1, x.shape)
 
-        return (1 / self._p_1) * mask.astype(x.dtype) * x
+        return (1 / self._p_1) * mask * x
+
+
+class Dropout2d(Module):
+    """Apply 2D channel-wise dropout during training.
+
+    Randomly zero out entire channels independently with probability p.
+    This layer expects input of shape (B, W, H, C) or (W, H, C).
```

Comment:
Could you add a comment like: 
```
where `B` is the batch dimension, `C` is the channel dimension, and `H` and `W` are the height and width respectively
```

Also I would add double backticks around the shapes.